PathOfBuildingCommunity/PathOfBuilding
# Allow https://poeskilltree.com Passive tree import

### Description of the problem being solved:
Cannot import https://poeskilltree.com/?v=3.17.0#AAAABAMAAFSufdLfig==
trees without copying and pasting the base64 into PoE's skilltree web page
 
### Steps taken to verify a working solution:
- Find a shed load of poeskilltree.com links by googling them and opening the parent page
- Open a build
- Select 'Import Tree'
- Paste a tree link and repeat as many times as needed
- Importing a build that is older than the current default version will properly load a spec of that version. Version older than 3.10 however cannot have more nodes assigned. This is a known fault.
![image](https://user-images.githubusercontent.com/4302241/154823861-1891f17f-b77d-437a-b862-e5785f43eaf7.png)


1. Uses decodeTreeLink function as per other imports, so if PoB can't import a tree version, then neither can this.
1. Doesn't check against atlas imports. They will fail silently in the same way a v1.0.0 tree does.
1. Has sanity checks to make sure the version on the url is a number and a valid tree version for PoB
1. No undo
2. Hijacked my own PR for fixing a fault in string:split

### Link to a build that showcases this PR:
None needed

### Before screenshot:
![image](https://user-images.githubusercontent.com/4302241/154783460-3469fe3f-0333-4585-b839-58aaf1c65865.png)


### After screenshot:
![image](https://user-images.githubusercontent.com/4302241/154783443-eb1c0595-f008-47d2-8f69-298427e970a3.png)


Alternative reference
https://www.reddit.com/r/pathofexile/comments/bufkuy/poeskilltreecom_historical_skill_trees/


Test links
https://www.pathofexile.com/passive-skill-tree/3.15.0/AAAABgMADI6-HwKSwQQHLJwtH9-wTLNfKoP3ES3r5AAA
https://poeskilltree.com/?v=3.6.0#AAAABAMAABEtfIOFMo6-ksHfsOvu
https://poeskilltree.com/passive-skill-tree/3.6.0/AAAABAMAABEtfIOFMo6-ksHfsOvu
http://poeurl.com/dABz (I have assumed that poeskilltree.com doesn't use these)



If you hit 'Convert Tree' before testing another link, you'll leave yourself a trail, so don't be confused by it, it was you who did it.
![image](https://user-images.githubusercontent.com/4302241/154824737-c867b388-01e3-4109-b06e-e690ff2c82fb.png)
